Config file updates

pazzarpj opened the pull request:
Config file updates to resolve #11 and #12.

Added ability to have default config path as <python_env>\fixate.yml
Added -c to specify config files
Added some config helper functions for loading yaml, loading config parameters from plugins/core modules.
Changed the default csv parameters to work with default launch parameters with no config files.

pazzarpj commented on src/fixate/config/helper.py:
```diff
@@ -60,3 +63,85 @@ def load_json_config(in_file, config_name=None):
+    if config_name:
+        fixate.config.__dict__.update({config_name: yaml.load(in_file)})
+    else:
+        fixate.config.__dict__.update(yaml.load(in_file))
```
This is just importing yaml file except I kept the conventional API that The json loader and dictionary loader that already existed in that file. The extra boiler plate is so you can pass a path instead of a file pointer

clint-lawrence commented:
There are some other bugs lurking here. If you call a script with no config set the timestamp doesn't get templated correctly. The colons with fail on windows, but creates the file fine on os x. However we open the log in append mode. And since the file name is no longer unique, it gets added into the single log file.

<img width="381" alt="screen shot 2018-10-05 at 12 33 40 am" src="https://user-images.githubusercontent.com/7078979/46481289-67cbd180-c836-11e8-9d48-0b9df8eb23a5.png">

clint-lawrence commented:
load_yaml_config() will fail (I think) if called with an open file handle. The os.path.exists() call should raise an exception.

pazzarpj commented:
> load_yaml_config() will fail (I think) if called with an open file handle. The os.path.exists() call should raise an exception.

Fixed by making it string only. if the string is valid file path, it will use the yaml library to open it, otherwise it will parse it as a string.

pazzarpj commented:
> There are some other bugs lurking here. If you call a script with no config set the timestamp doesn't get templated correctly. The colons with fail on windows, but creates the file fine on os x. However we open the log in append mode. And since the file name is no longer unique, it gets added into the single log file.
> 
> <img alt="screen shot 2018-10-05 at 12 33 40 am" width="381" src="https://user-images.githubusercontent.com/7078979/46481289-67cbd180-c836-11e8-9d48-0b9df8eb23a5.png">

Fixed the time date stamp not rendering with default config 
Moved tpl_time_stamp and tpl_csv_path outside of plg_csv so they are independently configurable and can be used in other places.

clint-lawrence commented on src/fixate/config/__init__.py:
```diff
@@ -29,11 +29,12 @@
 # Default settings for csv reporting. Can be configured via yaml either removing or overriding with plg_csv
+tpl_csv_path = ["{start_date_time}-{index}.csv"]
+tpl_time_stamp = "{0:%Y}{0:%m}{0:%d}-{0:%H}{0:%M}{0:%S}"
+
```
Doesn't need fixing now, but I think we need a log file name template and a log path. I'll raise an issue and we can tidy up later.

clint-lawrence replied:
Or maybe it's just need to be documented. See #21 for more thoughts
